pkg/oc/cli/cmd/login/loginoptions.go

```diff
@@ -316,16 +319,17 @@ func (o *LoginOptions) gatherProjectInfo() error {
 			return err
 		}
 		o.Project = current.Name
-
-		fmt.Fprintf(o.Out, "You have access to the following projects and can switch between them with '%s project <projectname>':\n\n", o.CommandName)
-		for _, p := range projects.List() {
-			if o.Project == p {
-				fmt.Fprintf(o.Out, "  * %s\n", p)
-			} else {
-				fmt.Fprintf(o.Out, "    %s\n", p)
+		if (!o.silentLogin()) {
+			fmt.Fprintf(o.Out, "You have access to the following projects and can switch between them with '%s project <projectname>':\n\n", o.CommandName)
```

Comment:
While limiting the output for general usage would be great, I don't feel it's a full solution to what my PR is trying to solve.

For those that are service providers and are logging in as cluster-admins or cluster-readers, they'll probably never end up using the first 20 projects so it's useless to print them out to them rather than just allowing them to disable the output of projects on login.

Reply:
If we added such a flag, I would expect this to print nothing and just have a non-zero exit code on failure.